### ISSUE TITLE
Customer Content-Type is being overwritten in createRequest in the BoxNetworkAgent

### ISSUE THREAD

**Issue by potrebic** [box/box-ios-sdk#840]
- [x] I have checked that the [SDK documentation][sdk-docs] doesn't solve my issue.
- [x] I have checked that the [API documentation][api-docs] doesn't solve my issue.
- [x] I have searched the [Box Developer Forums][dev-forums] and my issue isn't already reported (or if it has been reported, I have attached a link to it, for reference).
- [x] I have searched [Issues in this repo][github-repo] and my issue isn't already reported.

### Description of the Issue
I've run across an issue with the BoxSDK. in BoxNetworkAgent:
```
    private func createRequest(for request: BoxRequest) -> URLRequest {
        --- SNIP ---
        urlRequest.allHTTPHeaderFields = headers
```
So the headers are initialized with the headers passed in from the client. Good.
However, when composing the body the code will do things like the following:
```
     urlRequest.setValue("application/json", forHTTPHeaderField: "Content-Type")
```
And this will stomp the Content-Type that might have been set by the client.

### Expected Behavior
I expect the customer headers to be respected (aka not overwritten).

### Versions Used
iOS SDK: 5.2.0

[sdk-docs]: ./doc
[api-docs]: https://developer.box.com/docs
[dev-forums]: https://community.box.com/t5/Platform-and-Development-Forum/bd-p/DeveloperForum
[github-repo]: https://github.com/box/box-java-sdk/search?type=Issues

**Comment by lukaszsocha2** [comment]
Hi @potrebic,
customer Content-Type is being overwritten only when the specific body type is detected (json array, json object, url encoded form, multipart). So we overwrite it only when we are sure what the body content type really is. In other cases the customer header is passed without any change. Do you have any specific case when you need to send different Content-Type header than the actual content type of the body? 
Regards,
@lukaszsocha2

**Comment by arjankowski** [comment]
Fixed in #841
